Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -241,27 +241,35 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
     private func link(channels: [ChatChannel]) {
         guard !channels.isEmpty else { return }
         client.databaseContainer.write { session in
+            guard let queryDTO = session.channelListQuery(filterHash: self.query.filter.filterHash) else {
+                log.debug("Channel list query has not yet created \(self.query)")
+                return
+            }
+            
             for channel in channels {
                 guard let channelDTO = session.channel(cid: channel.cid) else {
                     log.error("Channel \(channel.cid) cannot be found in database.")
                     continue
                 }
-                let query = session.saveQuery(query: self.query)
-                query.channels.insert(channelDTO)
+                queryDTO.channels.insert(channelDTO)
             }
         }
     }
     
     private func unlink(channels: [ChatChannel]) {
         guard !channels.isEmpty else { return }
         client.databaseContainer.write { session in
+            guard let queryDTO = session.channelListQuery(filterHash: self.query.filter.filterHash) else {
+                log.debug("Channel list query has not yet created \(self.query)")
+                return
+            }
+            
             for channel in channels {
                 guard let channelDTO = session.channel(cid: channel.cid) else {
                     log.error("Channel \(channel.cid) cannot be found in database.")
                     continue
                 }
-                let query = session.saveQuery(query: self.query)
-                query.channels.remove(channelDTO)
+                queryDTO.channels.insert(channelDTO)
```

Comment:
Could we somehow add tests to make sure there are no regressions in the future?

Reply:
Used in the `guard` statements

Reply:
✅ Added tests